### ISSUE TITLE
Malformed RLP for NodeData message

### ISSUE THREAD

**Issue by Nashatyrev** [ethcore/parity#2887]
Receiving incorrectly formed `NodeData` message RLP from Parity (1.3.9):

`f904ea f870 9e 34988f87118b3e7a0b32ae23d46378c3425db42f214d24e164daeb36474f b84f f84d80896c6b935b8bbd400000a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421a0c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470 f869 9e 3198f8cdc716b7c6c5bc053c62a7913f92f250569d5d87b12757cbd50532 b848 f84602822710a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421a0c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470 6060604052361560275760e060020a60...`

The beginning of the message (3 node items) corresponds to the following items: 
- Trie node: `[34988f87118b3e7a0b32ae23d46378c3425db42f214d24e164daeb36474f, f84d80896c6b935b8bbd400000a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421a0c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470]`
- Trie node: `[3198f8cdc716b7c6c5bc053c62a7913f92f250569d5d87b12757cbd50532, f84602822710a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421a0c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470]`
- Code: `6060604052361560275760e060020a60...`

It seems the binary representation for each item is appended directly to the RLP without string length prefix which makes it impossible to parse the message (Trie node items still can be parsed by their RLP nature, but the Code item length is impossible to acquire) 

Geth node can't do fast sync with Parity as well: 
```
I1026 16:08:55.143867 eth/handler.go:252] Peer 48095a081b6d5959 [eth/63]: peer connected [Parity/v1.3.9-beta-e9987c4-20161021/x86_64-windows-msvc/rustc1.12.0]
I1026 16:08:56.267932 eth/downloader/downloader.go:320] Block synchronisation started
I1026 16:08:56.267932 eth/downloader/downloader.go:381] Synchronising with the network using: 48095a081b6d5959 [eth/63]
I1026 16:08:56.267932 eth/downloader/downloader.go:992] Peer 48095a081b6d5959 [hs 0.00/s, bs 0.00/s, rs 0.00/s, ss 0.00/s, miss    0, rtt 20s]: retrieving remote chain height
I1026 16:08:56.267932 eth/peer.go:252] Peer 48095a081b6d5959 [eth/63] fetching 1 headers from c85a1329, skipping 0 (reverse = false)
I1026 16:08:56.268932 eth/downloader/downloader.go:1040] Peer 48095a081b6d5959 [hs 0.00/s, bs 0.00/s, rs 0.00/s, ss 0.00/s, miss    0, rtt 20s]: looking for common ancestor
I1026 16:08:56.268932 eth/peer.go:259] Peer 48095a081b6d5959 [eth/63] fetching 192 headers from #0, skipping 0 (reverse = false)
I1026 16:08:56.293933 eth/downloader/downloader.go:1130] Peer 48095a081b6d5959 [hs 0.00/s, bs 0.00/s, rs 0.00/s, ss 0.00/s, miss    0, rtt 20s]: common ancestor: #0 [d4e56740…]
I1026 16:08:56.293933 eth/downloader/downloader.go:460] Fast syncing until pivot block #743981
I1026 16:08:56.294933 eth/downloader/downloader.go:1406] Downloading node state data
I1026 16:08:56.294933 eth/downloader/downloader.go:1358] Downloading block bodies from #1
I1026 16:08:56.294933 eth/downloader/downloader.go:1382] Downloading receipts from #1
I1026 16:08:56.294933 eth/downloader/downloader.go:1211] Peer 48095a081b6d5959 [hs 0.00/s, bs 0.00/s, rs 0.00/s, ss 0.00/s, miss    0, rtt 20s]: directing header downloads from #1
I1026 16:08:56.294933 eth/peer.go:259] Peer 48095a081b6d5959 [eth/63] fetching 128 headers from #192, skipping 191 (reverse = false)
I1026 16:08:56.310934 eth/downloader/downloader.go:1327] Filling up skeleton from #1
I1026 16:08:56.394939 eth/peer.go:273] Peer 48095a081b6d5959 [eth/63] fetching 1 state data
I1026 16:08:56.394939 eth/handler.go:284] Peer 48095a081b6d5959 [eth/63]: message handling failed: Invalid message - msg msg #14 (535 bytes): invalid message: (code e) (size 535) rlp: expected input string or byte for []uint8, decoding into ([][]uint8)[0]
```

**Comment by keorn** [comment]
The RLP is malformed, so its a bug if it comes from Parity.

**Comment by arkpar** [comment]
Should be fixed in #2892

**Comment by Nashatyrev** [comment]
Thanks!
